libr/egg/p/sc/asm/x86-osx-binsh.rasm

```diff
@@ -0,0 +1,46 @@
+; x86_64 macOS execve /bin/sh shellcode with setuid(0)
+; Assembly equivalent of the bytes in libr/egg/p/sc/src/x86-osx-binsh.c
+; 
+; This shellcode first calls setuid(0) and then execve("/bin/sh", argv, NULL)
+; Uses jmp/call technique for position-independent string address
+;
+; Note: Using .byte directives for encoding-sensitive instructions
+; to match original bytes exactly (rasm2 may produce different encodings)
+
+; mov r8b, 2 - Setup syscall base
+.byte 0x41
```

Comment:
if any of those instructions isnt assembled properly we may want to improve r2 assemblers to make them work. lets think about this task as a way to improve rasm2